compiler/vm/vmgen.nim

```diff
@@ -1371,9 +1371,20 @@ proc genCard(c: var TCtx; n: PNode; dest: var TDest) =
   c.gABC(n, opcCard, dest, tmp)
   c.freeTemp(tmp)
 
+func fitsRegisterConsiderView(t: PType): bool =
+  ## Returns whether a value of type `t` fits into a register, also
+  ## considering view-types that map to pointers.
+  # XXX: introduce a ``mapType`` (similar to the one used by the other code
+  #      generators) and base the "fits register" queries on that
+  let t = t.skipTypes(IrrelevantTypes)
+  result = fitsRegister(t)
+  if not result:
+    # is it a direct single-location view?
+    result = t.kind in {tyVar, tyLent} and t.base.kind != tyOpenArray
```

Comment:
```suggestion
fitsRegister(t) or
    t.kind in {tyVar, tyLent} and t.base.kind != tyOpenArray
    # above tests for direct single-location view
```

very soft suggestion, it's slightly less imperative by exploiting mcarthy evaluation; however, fitting in the comment neatly is tricky.

Reply:
I changed it into an expression, but put the comment (with a colon at the end) above the second condition.